src/__tests__/wrap.v2.e2e.test.ts

```diff
@@ -0,0 +1,56 @@
+import { handler } from "../commands/wrap";
+import path from "path";
+import tmp from "tmp";
+import signale from "signale";
+
+const fixtureFolderName = path.join("fixture", "2.0");
+const validFileName = "example.0.json";
+const exampleFolderPath = path.resolve("examples", "v2", "raw-documents", validFileName);
+const validFileNameWithCustomSchemaWithNestedRef = "valid-custom-schema-with-nested-ref-document.json";
+const validFileNameWithCustomSchema = "valid-custom-schema-document.json";
+
+describe("wrap with file input", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  const signaleSuccessSpy = jest.spyOn(signale, "success");
+  describe("without custom schema", () => {
+    it("should allow output when given valid open attestation document", async () => {
+      const outputFile = tmp.fileSync();
+      await handler({
+        rawDocumentsPath: exampleFolderPath,
+        outputFile: outputFile.name,
+        openAttestationV3: false,
+        unwrap: false,
+        batched: true,
+      });
+      expect(signaleSuccessSpy).toHaveBeenCalledTimes(1);
+    });
+  });
+  describe("with custom schema", () => {
+    it("should allow output as file if input path is an input file with custom schema", async () => {
+      const outputFile = tmp.fileSync();
+      await handler({
+        rawDocumentsPath: path.resolve(__dirname, fixtureFolderName, validFileNameWithCustomSchema),
+        outputFile: outputFile.name,
+        schema: path.resolve(__dirname, fixtureFolderName, "custom-schema.json"),
+        openAttestationV3: false,
+        unwrap: false,
+        batched: true,
+      });
+      expect(signaleSuccessSpy).toHaveBeenCalledTimes(1);
+    });
+    it("should allow output as file if input path is an input file with custom schema that has a reference to an external schema", async () => {
+      const outputFile = tmp.fileSync();
+      await handler({
+        rawDocumentsPath: path.resolve(__dirname, fixtureFolderName, validFileNameWithCustomSchemaWithNestedRef),
+        outputFile: outputFile.name,
+        schema: path.resolve(__dirname, fixtureFolderName, "custom-schema-with-nested-ref.json"),
+        openAttestationV3: false,
+        unwrap: false,
+        batched: true,
+      });
+      expect(signaleSuccessSpy).toHaveBeenCalledTimes(1);
```

Comment:
Maybe we should add additional verification on the output file instead:
The output file must exist and must have a signature with merkle root and target hash. The value of both doesn't matter at long as it exists

Wdyt ?

Reply:
yup that would be good, (i initially did it that way, but thought it was an e2e test so opted for the more blackbox way).

i'll add it in.